pkg/controllers/scalablenodegroup/v1alpha1/suite_test.go

```diff
@@ -53,24 +54,87 @@ var _ = AfterSuite(func() {
 
 var _ = Describe("Examples", func() {
 	var ns *environment.Namespace
+	var desiredReplicas = ptr.Int32(5)
+	var dummyMessage = "test message"
+	var nodeGroupObject *fake.NodeGroup
 	var sng *v1alpha1.ScalableNodeGroup
+	var ng cloudprovider.NodeGroup
+	var defaultNodeGroupObject fake.NodeGroup
 
 	BeforeEach(func() {
```

Comment:
I'm having a little trouble following the flow of the BeforeEach, Test, AfterEach logic. It looks like some logic is duplicated (i.e. sng.Spec.Replicas being set). 

When I write these, I typically try to do everything in a BeforeEach loop, unless my test has mutated something that needs to be cleaned up after all of the tests.

I also think it might be cleaner to modify the fields of the CloudProvider factory, rather than using a singleton pattern on nodegroup and modifying that object directly. I think all you'd have to do is wire through the behavior of NodeGroupStabilized and NodeGroupStabilizedMessage to the CloudProvider factory, and then before each test, you can just do something like
```
factory.NodeGroupStabilized = false 
factory.NodeGroupStabilizedMessage = "still trying"
```
Then when the factory creates the object, it will inject the behavior into the node group and should clean up most of your singleton/casting logic.

Let me know what you think to this approach.